rsocket-core/src/main/java/io/rsocket/resume/InMemoryResumeStore.java

```diff
@@ -0,0 +1,113 @@
+package io.rsocket.resume;
+
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoProcessor;
+import reactor.util.concurrent.Queues;
+
+import java.util.Queue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class InMemoryResumeStore implements ResumeStore {
+  private static final Logger logger = LoggerFactory.getLogger(InMemoryResumeStore.class);
+
+  private final MonoProcessor<Void> disposed = MonoProcessor.create();
+  private final AtomicLong position = new AtomicLong();
+  private final AtomicLong impliedPosition = new AtomicLong();
+  private volatile AtomicInteger cachedFramesSize = new AtomicInteger();
+  private final Queue<ByteBuf> cachedFrames;
+  private final String tag;
+  private final int cacheLimit;
+
+  public InMemoryResumeStore(String tag, int cacheLimit) {
+    this.tag = tag;
+    this.cacheLimit = cacheLimit;
+    this.cachedFrames = cachedFramesQueue(cacheLimit);
+  }
+
+  @Override
+  public void saveFrame(ByteBuf frame) {
+    cachedFramesSize.incrementAndGet();
+    cachedFrames.offer(frame.retain());
```

Comment:
add seems better than offer considering you don't check return value.

Reply:
Queue implementation (`SpscArrayQueue`) does not support that method. I can check return value and throw if queue is full.  Also, because `saveFrame` wont be called concurrently (access is serialized by calling code), current logic looks sufficient . Another thing - looks like atomic is not needed here as `cachedFramesSize` (and other fields except `impliedPosition`) due to above - just should be safe to access from different threads